src/main/java/com/znsio/teswiz/runner/BrowserStackImageInjection.java

```diff
@@ -0,0 +1,64 @@
+package com.znsio.teswiz.runner;
+
+import com.znsio.teswiz.tools.cmd.CommandLineExecutor;
+import com.znsio.teswiz.tools.cmd.CommandLineResponse;
+import io.appium.java_client.AppiumDriver;
+import org.apache.commons.lang3.NotImplementedException;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.util.Arrays;
+import org.apache.log4j.Logger;
+import static com.znsio.teswiz.runner.Setup.*;
+import static com.znsio.teswiz.tools.Wait.waitFor;
+
+public class BrowserStackImageInjection {
+
+    private static final Logger LOGGER = Logger.getLogger(BrowserStackImageInjection.class.getName());
+
+
+    private static String uploadToCloud(String uploadFilePath) {
+        uploadFilePath = new File(uploadFilePath).getAbsolutePath();
+        String fileName = new File(uploadFilePath).getName();
+        String cloudUser = Setup.getFromConfigs(Setup.CLOUD_USER);
+        String cloudKey = Setup.getFromConfigs(CLOUD_KEY);
+        String cloudName = Setup.getFromConfigs(CLOUD_NAME);
+        String mediaUrl = Runner.NOT_SET;
+
+        if (cloudUser == null && cloudKey == null && cloudName == null) {
+            cloudUser = System.getenv(CLOUD_USER);
+            cloudKey = System.getenv(CLOUD_KEY);
+            cloudName = System.getenv(CLOUD_NAME);
+        }
+
+        if (cloudName.equalsIgnoreCase("browserstack")) {
```

Comment:
Same for this `if` condition check. This will always be in the context of BrowserStack only.
Maybe we should not have the `injectMediaToDriver` public. We should create a method in `Driver` - `injectMediaToDriver` - in that method, we should throw a method not supported if 
- platform is not android, AND
- CLOUD_NAME is not BrowserStack
Only if the above conditions pass, then call the `injectMediaToDriver` method. 

This way, the users continue to work with the Driver object.